src/main/scala/li/cil/oc/util/FluidUtils.scala

```diff
@@ -44,15 +44,21 @@ object FluidUtils {
     val ti = source.getTankInfo(sourceSide)
     val srcFluid = if (sourceTank < 0 || ti == null || ti.length <= sourceTank) null else ti(sourceTank).fluid.copy()
 
-    val drained = if (srcFluid == null)
+    val nullFluid = srcFluid == null;
+    val drained = if (nullFluid)
       source.drain(sourceSide, limit, false)
     else {
       srcFluid.amount = limit
       source.drain(sourceSide, srcFluid, false)
     }
     if (drained != null) {
       val filled = sink.fill(sinkSide, drained, false)
-      sink.fill(sinkSide, source.drain(sourceSide, filled, true), true)
```

Comment:
I think I meant 'drained' instead of 'filled' here. Having 2 signatures hidden the mistake.